packages/@webex/plugin-meetings/test/unit/spec/recording-controller/index.js

```diff
@@ -0,0 +1,284 @@
+import RecordingController from '@webex/plugin-meetings/src/recording-controller';
+import sinon from 'sinon';
+import {assert} from '@webex/test-helper-chai';
+import { HTTP_VERBS } from '@webex/plugin-meetings/src/constants';
+
+describe('plugin-meetings', () => {
+    describe('recording-controller tests', () => {
+        describe('index', () => {
+            let request;
+
+            describe('class tests', () => {
+                it('can create an instance with the extracted values', () => {
+                    const controller = new RecordingController({}, {
+                        url: 'test/id',
+                        fullState: {
+                            sessionId: 'testId',
+                        },
+                        services: {
+                            record: {
+                                url: 'test'
+                            }
+                        }
+                    })
+                    assert(controller.serviceUrl, 'test');
```

Comment:
we're reading private properties of the class here - I would say this is not ideal, but I leave it up to you, it will be a bit more work to test this if you can't read these properties. If this file was a ts file not js, you would have typescript errors here. Anyway, also worth considering if maybe the tests should be ts files? Anyway, up to you.

Reply:
yeah, I'll add getters and test using it that way